datadog/resource_datadog_powerpack.go

```diff
@@ -707,37 +735,113 @@ func normalizeDashboardWidgetDef(widgetDef map[string]interface{}, columnWidth i
+	}
 	return widgetDef, diags
 }
 
 func normalizeTerraformWidgetDef(widgetDef map[string]interface{}) map[string]interface{} {
 	// Dashboard TF widgets have a "requests" field, while API Spec has a "request" field
 	// Here we set the "request" field and remove "requests"
 	if widgetDef["requests"] != nil {
-		widgetDefRequests := widgetDef["requests"].([]interface{})
-		for i, widgetDefRequest := range widgetDefRequests {
-			widgetDefRequestNormalized := widgetDefRequest.(map[string]interface{})
-			if widgetDefRequestNormalized["limit"] != nil {
-				widgetDefRequestNormalized["limit"] = int(widgetDefRequestNormalized["limit"].(float64))
+		if widgetDef["type"] == "scatterplot" || widgetDef["type"] == "hostmap" {
+			widgetDefRequest := widgetDef["requests"].(map[string]interface{})
+			for _, v := range []string{"fill", "size", "x", "y"} {
+				// Properties listed above need to be converted from single values in the API to plural values for TF
+				if widgetDefRequest[v] != nil {
+					widgetDefRequest[v] = []interface{}{widgetDefRequest[v].(interface{})}
+				}
 			}
-			widgetDefRequests[i] = widgetDefRequestNormalized
+			widgetDef["request"] = []interface{}{widgetDefRequest}
+		} else {
+			widgetDefRequests := widgetDef["requests"].([]interface{})
+			for i, widgetDefRequest := range widgetDefRequests {
+				widgetDefRequestNormalized := widgetDefRequest.(map[string]interface{})
+				for _, v := range []string{"style", "apm_stats_query"} {
+					// Properties listed above need to be converted from single values in the API to plural values for TF
+					if widgetDefRequestNormalized[v] != nil {
+						widgetDefRequestNormalized[v] = []interface{}{widgetDefRequestNormalized[v].(interface{})}
+					}
+				}
+				if widgetDefRequestNormalized["limit"] != nil {
+					// Dashboard widget can't typecast the float64 limit value so we need to convert it first
+					widgetDefRequestNormalized["limit"] = int(widgetDefRequestNormalized["limit"].(float64))
+				}
+				if widgetDef["type"] != "topology_map" && widgetDefRequestNormalized["query"] != nil {
+					widgetDefRequestNormalized["query"] = []interface{}{widgetDefRequestNormalized["query"].(interface{})}
+				}
+				widgetDefRequests[i] = widgetDefRequestNormalized
+			}
+			widgetDef["request"] = widgetDefRequests
 		}
-		widgetDef["request"] = widgetDefRequests
 		delete(widgetDef, "requests")
 	}
+	for _, v := range []string{"style", "xaxis", "yaxis", "y"} {
+		// Properties listed above need to be converted from single values in the API to plural values for TF
+		if widgetDef[v] != nil {
+			widgetDef[v] = []interface{}{widgetDef[v].(map[string]interface{})}
+		}
+	}
+	for _, v := range []string{"custom_links", "inputs", "events"} {
+		// Properties listed above are defined as plural for the API Spec and need to be converted back to single values for TF
+		if widgetDef[v] != nil {
+			widgetDef[v[:len(v)-1]] = widgetDef[v]
+			delete(widgetDef, v)
+		}
+	}
```

Comment:
**suggestion(non-blocking):** Similar to a suggestion above, we may want to consider remapping properties w/ some kinda map here, rather than relying the on the "s"-suffix relation